File: src/main/java/org/pl/parser/ast/UnaryOperationNode.java
package org.pl.parser.ast;

public class UnaryOperationNode implements INode {
    public final INode node;
    public final UnaryOperation op;

    public UnaryOperationNode(INode node, UnaryOperation op) {
        this.node = node;
        this.op = op;
    }

    @Override
    public String toString() {
        return "UnaryOperationNode{" +
                "node=" + node +
                ", op=" + op +
                '}';
    }
}

File: src/main/kotlin/org/leaflang/parser/ast/UnaryOperationNode.kt
package org.leaflang.parser.ast

import org.leaflang.parser.utils.NodePosition

/**
 * A unary operation node that contains some operand ([node]) and an operator ([op]).
 */
class UnaryOperationNode(override val position: NodePosition,
                         val node: INode,
                         val op: UnaryOperation) : INode {
    override fun toString() = "UnaryOperationNode{node=$node, op=$op}"
}